### ISSUE TITLE
Cannot delete specific group

### ISSUE THREAD

**Issue by manuasir** [wazuh/wazuh#3970]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
|3.9.5 | Framework| Manager| Packages | Amazon Linux 2 |

Hello team,

### Description

Currently, there is a bug in the `DELETE /agents/groups/` API endpoint. It seems to happen only when using the specific `group` name.

### How to reproduce

1. Create a group named `group`.
```
PUT /agents/groups/group
```

The result of that request is fine. No errors.

2. Remove the group

```
DELETE /agents/groups/group
```

The result is the following error

```
{"message":"Wazuh API error: 600 - Param not valid. Integer expected.  Field: agent_id","code":"Wazuh API error: 600","statusCode":500}
```

### Graphic description

![cant delete group agents](https://user-images.githubusercontent.com/10210567/64878324-9b70c780-d653-11e9-8812-cbd6e58b0446.gif)

Regards

**Comment by davidjiglesias** [comment]
Hello team,

The problem here is simple, the API call to delete groups is:

`DELETE /agents/groups/:group_id`

And the call to remove agents from all groups is:

`DELETE /agents/:agent_id/group`


So, when attempting to delete a group called `group`,  the query is:

`DELETE /agents/groups/group`, which matches both endpoints, in the first case it means to delete group called `group`, but in the second case it means to delete agent with id `group` from all groups.

It looks like the second endpoint is matched first and therefore we get this error.

The problem here is a structural one, since the request to delete the group should not be under /agents, but rather a new route /groups, they operate with two different wazuh resources.

**Comment by davidjiglesias** [comment]
Fixed for new API in #4571
